internal/route/node_filter.go

```diff
@@ -12,28 +12,99 @@ type NodeFilter interface {
 	Apply(requestMetadata *RequestMetadata, upstreamConfig *config.UpstreamConfig) bool
 }
 
-type IsHealthyAndAtGlobalMaxHeightFilter struct {
+type AndFilter struct {
+	filters []NodeFilter
+}
+
+func (a *AndFilter) Apply(requestMetadata *RequestMetadata, upstreamConfig *config.UpstreamConfig) bool {
+	var result = true
+
+	for filterIndex := range a.filters {
+		var filter = a.filters[filterIndex]
+		if !filter.Apply(requestMetadata, upstreamConfig) {
+			result = false
+			break
+		}
+	}
+
+	return result
+}
+
+type IsHealthyFilter struct {
```

Comment:
nit: do we need `Filter` in the name? The other structs don't have it.

Reply:
Or maybe just `HealthyFilter`? I feel like this `Is` prefix might be a little verbose. Indeed it's the general naming convention for `boolean` methods but I feel like the intent is already communicated clearly without the `Is` prefix.